var/spack/repos/builtin/packages/py-partd/package.py

```diff
@@ -12,11 +12,13 @@ class PyPartd(PythonPackage):
     homepage = "https://github.com/dask/partd/"
     pypi = "partd/partd-0.3.8.tar.gz"
 
+    version("1.4.0", sha256="aa0ff35dbbcc807ae374db56332f4c1b39b46f67bf2975f5151e0b4186aed0d5")
     version("1.1.0", sha256="6e258bf0810701407ad1410d63d1a15cfd7b773fd9efe555dac6bb82cc8832b0")
     version("0.3.10", sha256="33722a228ebcd1fa6f44b1631bdd4cff056376f89eb826d7d880b35b637bcfba")
     version("0.3.8", sha256="67291f1c4827cde3e0148b3be5d69af64b6d6169feb9ba88f0a6cfe77089400f")
 
     depends_on("python@3.5:", type=("build", "run"), when="@1.1.0:")
+    depends_on("python@3.7:", type=("build", "run"), when="@1.4.0:")
```

Comment:
Spack only supports Python 3.7+ so you could technically remove both lines. Not required though